alembic/ddl/postgresql.py

```diff
@@ -512,6 +516,34 @@ def __init__(
         self.using = using
 
 
+@compiles(AddColumn, "postgresql")
+def visit_add_column(element: AddColumn, compiler: PGDDLCompiler, **kw) -> str:
+    return "%s %s" % (
+        alter_table(compiler, element.table_name, element.schema),
+        add_column(
```

Comment:
why not just pass through if_exists / if_not_exists in base.py? it's just an argument, the compiler can still ignore it if it chooses. i dont think we need new AddColumn / DropColumn @compiles here

Reply:
Given that alembic can be used to apply migration on several servers, my primary concern was that the automated generation of `if_exist` would possibly break for someone using a combination of postgresql and something else.

However you master way more than me the different usages of alembic and if you feel confident that we can move it all to `base.py` and remove it from implementations, I'd be happy to do so.
(It'll also solve most of the other questions)

Reply:
but to be clear, just do the first part of that paragraph unless you have a great idea for the second part.  otherwise Ill take the succinct version of the code and figure something out.